src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.LocalFunctionReferenceRewriter.cs

```diff
@@ -76,6 +98,70 @@ public override BoundNode VisitConversion(BoundConversion conversion)
             }
         }
 
+        /// <summary>
+        /// Visit all references to local functions (calls, delegete
+        /// conversions, delegate creations) and rewrite them to point
+        /// to the rewritten local function method instead of the original. 
+        /// </summary>
+        public BoundStatement RewriteLocalFunctionReferences(BoundStatement loweredBody)
+        {
+            var rewriter = new LocalFunctionReferenceRewriter(this);
+
+            Debug.Assert(_currentMethod == _topLevelMethod);
+
+            // Visit the body first since the state is already set
+            // for the top-level method
+            var newBody = (BoundStatement)rewriter.Visit(loweredBody);
+
+            // Visit all the rewritten methods as well
+            var synthesizedMethods = CompilationState.SynthesizedMethods;
+            if (synthesizedMethods != null)
+            {
+                // Dump the existing methods for rewriting
+                var oldMethods = synthesizedMethods.ToImmutable();
+                synthesizedMethods.Clear();
```

Comment:
Isn't `CompilationState.SynthesizedMethods` shared across all methods for the enclosing `NamedTypeSymbol`?